src/main/java/org/codehaus/plexus/util/VersionSpecifics.java

```diff
@@ -0,0 +1,31 @@
+package org.codehaus.plexus.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.io.Writer;
+
+/**
+ * Implementation specific to Java SE 8 version.
+ */
+final class VersionSpecifics
```

Comment:
I still don't like the classname. It should contain something with IO, because other `o.c.p.util` classes might need a version specific implementation too, that shouldn't be mixed with this one.

Reply:
In fact I do not understand why you do not want to mix it. Having all tweaks in a single class makes it rather easy to find all the version specifics very easily.

Reply:
Suppose we can have improvements by using TextBlock for some StringUtils. We would need to introduce a new VersionSpecifics, and it will contain clones of current copy methods, while existing VersionSpecifics need to be extended with the newly introduces String methods.
Maintaining MRJAR code is already hard, this will even make it harder.

Reply:
There is actually no need for "a new VersionSpecifics". The idea behind my code actually is that Plexus-Utils JUST has that single `VersionSpecifics` class for *all* packages, so maintaining the MRJARs will become a single-location task, hence is *pretty easy* to do: Just put a new `static` **method** in the existing `VersionSpecifics`, and make the original code call it. I cannot see what is complex with that, and in fact, what you counter-propose makes it much, much more complex, as you will have *lots* of version specific classes in the end, *and* nobody would find them as it is totally unclear why the implementation of version specifics should be named "util", as these are not utilities *at all* but actually are what they are: version specifics. Sorry, I cannot see your point here at all.